File: gerencia_de_estoque/src/main/Filial.java
package main;
import java.util.ArrayList;
import java.util.Iterator;

public class Filial {
	private String nome;
	private String local;
	private int id;
	private ArrayList<Item> itens;
	
	public Filial(String nome, String local, int id) {
		this.nome = nome;
		this.local = local;
		this.id = id;
		itens = new ArrayList<Item>();
	}
	
	public void adicionarItem(Item i) {
		// Adds an item to branch item list
		itens.add(i);
	}
	
	public void removerItem(int itemId) {
		// Removes an item from branch item list, given its ID
		Iterator<Item> itr = itens.iterator();
		
		while (itr.hasNext()) {
			Item currentItem = itr.next();
			
			if (currentItem.getId() == itemId)
				itr.remove();
		}
	}
	
	public Item getItem(int id) {
		// Returns an specific item given its ID
		// returns null if item is not found
		for (Item item : itens) {
			if (item.getId() == id) {
				return item;
			}
		}
		return null;
	}

	public ArrayList<Item> getAllItens() {
		// Returns an item list of this branch
		return itens;
	}
	
	public ArrayList<Item> getEstoqueVazio() {
		// Returns a list of all items out of stock
		ArrayList<Item> itensVazios = new ArrayList<Item>();
		for (Item item : itens) {
			if (item.getQuantidade() == 0) {
				itensVazios.add(item);
			}
		}
		return itensVazios;
	}
	
	public String toString() {
		return String.format("---Filial---\nid: %d\nnome: %s\nlocal: %s\n", id, nome, local);
	}
	
	public String getNome() {
		return nome;
	}

	public void setNome(String nome) {
		this.nome = nome;
	}

	public String getLocal() {
		return local;
	}

	public void setLocal(String local) {
		this.local = local;
	}

	public int getId() {
		return id;
	}

	public void setId(int id) {
		this.id = id;
	}

}

File: estoque/app/model/filial.py
from model.item import Item


class Filial:
    def __init__(self, nome: str, local: str, id: int):
        self._nome: str = nome
        self._local: str = local
        self._id: int = id
        self._itens: list[Item] = []

    def adicionar_item(self, item: Item) -> None:
        self._itens.append(item)

    def remover_item(self, item: Item) -> None:
        if item in self._itens:
            self._itens.remove(item)

    def listar_caracteristicas(self) -> str:
        return f"ID: {self._id}\nNome: {self._nome}\nLocal: {self._local}\nQtd Itens: {len(self._itens)}"

    @property
    def itens(self) -> list[Item]:
        return self._itens.copy()

    @property
    def nome(self) -> str:
        return self._nome

    @nome.setter
    def nome(self, nome: str) -> None:
        self._nome = nome

    @property
    def local(self) -> str:
        return self._local

    @local.setter
    def local(self, local: str) -> None:
        self._local = local

    @property
    def id(self) -> int:
        return self._id

    @id.setter
    def id(self, id: int) -> None:
        self._id = id

    def __str__(self) -> str:
        return f"Filial {self._nome} (ID: {self._id}) - Local: {self._local}"
